### PR TITLE
Added class to initialize ip thermal camera

### PR DESCRIPTION
**Pull request by Janek3d**
Class contains setting to establish connection with camera througth rtps protocol and also function to get image from that camera.

### CODE REVIEW

File: IoT_master/tools/camera_initialize.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import cv2
+
+class Camera:
+    def __init__(self):
+        self.login = 'admin'
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: Class variables shouldn’t be hardcoded. Please add them as function arguments.

File: IoT_master/tools/camera_initialize.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        self.password = 'qwerty123456'
+        self.ip_address = '192.168.1.108'
+        self.channel = '1'
+        self.adres_part_1 = 'rtsp://'+self.login+':'+self.password+'@'+self.ip_address+'/cam/realmonitor?channel='
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: Does it really have to be specified here?
Move it maybe to separate static method that will generate appropriate string based on arguments passed.

Also, I see that probably this line is too long to pass tests.
Have you run `make test` command on your code?

File: IoT_master/tools/camera_initialize.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import cv2
+
+class Camera:
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: Can you please add docstring on what this class does?
It should be formatted:
```
“””Short description on what this class does
and what is it’s purpose.
“””
```

File: IoT_master/tools/camera_initialize.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    def __init__(self):
+        self.login = 'admin'
+        self.password = 'qwerty123456'
+        self.ip_address = '192.168.1.108'
+        self.channel = '1'
+        self.adres_part_1 = 'rtsp://'+self.login+':'+self.password+'@'+self.ip_address+'/cam/realmonitor?channel='
+        self.adres_part_2 = '&subtype=0'
+        
+    def camera(self,channel='thermal'):
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: Can you please add docstrings to your functions?
The format should be:
```
“””Short description of function’s purpose
with all necessary details for anyone
that will read this code in the future.

:param myvar: Description of myvar
:type myvar: Type of myvar
:param myvar2: Description of myvar2
:type myvar2: type of myvar2

:return: Description of what is returned
:rtype: returned type
“””
```

File: IoT_master/tools/camera_initialize.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import cv2
+
+class Camera:
+    def __init__(self):
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: Can you please add docstrings to your functions?
The format should be:
```
“””Short description of function’s purpose
with all necessary details for anyone
that will read this code in the future.

:param myvar: Description of myvar
:type myvar: Type of myvar
:param myvar2: Description of myvar2
:type myvar2: type of myvar2

:return: Description of what is returned
:rtype: returned type
“””
```

However, don’t describe self parameter

File: IoT_master/tools/camera_initialize.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            self.adres = self.adres_part_1+'1'+self.adres_part_2
+        self.cam = cv2.VideoCapture(self.adres)
+        self.ret, self.img = self.cam.read()
+        return self.ret, self.img
+
+if __name__=='__main__':
+
+    kamera_class = Camera()
+    cv2.namedWindow('Camera')
+    #cv2.namedWindow('Camera')
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: If it’s not necessary for script then remove it

File: IoT_master/tools/camera_initialize.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    def camera(self,channel='thermal'):
+        if channel == 'thermal':
+            self.adres = self.adres_part_1+'2'+self.adres_part_2
+        else:
+            self.adres = self.adres_part_1+'1'+self.adres_part_2
+        self.cam = cv2.VideoCapture(self.adres)
+        self.ret, self.img = self.cam.read()
+        return self.ret, self.img
+
+if __name__=='__main__':
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: Can you please add docstring that will provide some info of what is going on when you run this script?
I know what it does, but it’s be helpful for other people

File: IoT_master/tools/camera_initialize.py (hunk 3 above)
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py]: I’m not sure whether `Camera` speaks for itself. Maybe `ThermalCamera` would be more meaningful

**Review by Dysproz — CHANGES_REQUESTED** [review]

File: IoT_master/tools/camera_initialize.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import cv2
```
Comment (Dysproz) [inline IoT_master/tools/camera_initialize.py:1]: Add OpenCV to requirements.txt also

**Review by Dysproz — CHANGES_REQUESTED** [review]

File: IoT_master/tools/camera_initialize.py (hunk 1 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: Now reads values from ini file.

File: IoT_master/tools/camera_initialize.py (hunk 2 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: I have run tox test if that's what you mean.

File: IoT_master/tools/camera_initialize.py (hunk 3 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: Done.

File: IoT_master/tools/camera_initialize.py (hunk 4 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: Done.

File: IoT_master/tools/camera_initialize.py (hunk 5 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: Done.

File: IoT_master/tools/camera_initialize.py (hunk 6 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: Removed.

File: IoT_master/tools/camera_initialize.py (hunk 7 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py]: Done.

File: IoT_master/tools/camera_initialize.py (hunk 8 above)
Comment (Janek3d) [reply IoT_master/tools/camera_initialize.py:1]: Added.

**Review by Dysproz — APPROVED** [review]

**Pull request merged**